### PR TITLE
 fix(queue): function `Queue.can_enqueue` should handle non-existing queue instance correctly

### PR DESCRIPTION
**Pull request by ADD-SP**
### Summary

Treating the non-existing queue instance has enough capacity for enqueueing (fixup https://github.com/Kong/kong/pull/13164)

### Checklist

- [X] The Pull Request has tests
- [N/A] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [N/A] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

_[KAG-4270]_

[KAG-4270]: https://konghq.atlassian.net/browse/KAG-4270?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kong/tools/queue.lua [hunk 1]
```diff
@@ -94,9 +94,17 @@ local Queue_mt = {
   __index = Queue
 }
 
+-- string buffer for constructing queue keys efficiently
+local buf = require("string.buffer").new(32)
+
 
 local function _make_queue_key(name)
-  return (workspaces.get_workspace_id() or "") .. "." .. name
+  local key = buf:put(
```
Comment (chronolaw) [inline kong/tools/queue.lua:102]: Do we have evidence that for this short key `string.buffer` is faster than concat?

Comment (ADD-SP) [reply kong/tools/queue.lua]: No, the buffer is much slower, reverted.

File: kong/tools/queue.lua [hunk 2]
```diff
@@ -297,8 +297,18 @@ end
...
+    --          this function will incorrectly return `false` instead of `true`.
+    --          This is a limitation of the current implementation.
+    --          All capacity checking functions needs a Queue instance to work correctly.
+    --          consturcting a Queue instance just for this function is not efficient,
+    --          so we just return `true` here.
+    --          This limitation should not happen in normal usage,
+    --          as user should be aware of the queue capacity settings
+    --          to avoid such situation.
+    return true
   end
```
Comment (chobits) [inline kong/tools/queue.lua:312]: Cool. changing this return value is safe for kong, cause currently there is no users calling this method in kong-ce.

**Review by chobits — APPROVED** [review]

File: kong/tools/queue.lua [hunk 3]
```diff
@@ -297,8 +297,18 @@ end
...
   local queue = queues[_make_queue_key(queue_conf.name)]
   if not queue then
-    -- treat non-existing queues as not full as they will be created on demand
-    return false
+    -- treat non-existing queues having enough capacity.
+    -- WARNING: The limitation is that if the `entry` is a string and the `queue.max_bytes` is set,
+    --          and also the `#entry` is larger than `queue.max_bytes`,
+    --          this function will incorrectly return `false` instead of `true`.
+    --          This is a limitation of the current implementation.
+    --          All capacity checking functions needs a Queue instance to work correctly.
```
Comment (chronolaw) [inline kong/tools/queue.lua:305]:
```suggestion
    --          All capacity checking functions need a Queue instance to work correctly.
```

Comment (ADD-SP) [reply kong/tools/queue.lua]: fixed

**Review by chronolaw — APPROVED** [review]

**Pull request merged**